Automattic/jetpack
# CRM: Add webpack build step

In the CRM Gulp is used to build assets (minify JS, compile SASS into CSS). 
This PR aims to change this from Gulp to webpack.
We are doing this because there is an effort to remove gulp from the Monorepo (p9dueE-4RS-p2).

## Proposed changes:

* Adds webpack build step to the CRM
* Removes gulp 
* Hides built files via .gitignore and defines which files to include in production in .gitattributes.
* Includes some changes to existing CSS and min files to make compilation and minification smoother.

### Other information:


## Jetpack product discussion


p1672845316058569-slack-C04J4UG3FLG

## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Spin up a version of this branch
* Run `jetpack build plugins/crm` for a development build, and `jetpack build --production plugins/crm` to test the build as if it were in a production environment. 
* For each JS file inside a `js` folder, a minified JS file named `[original filename].min.js` should be created (in the same folder) (actual minification only happens if running a production build).
* For each SCSS file inside a `sass` folder, a minified CSS file named `[original filename].min.css` should be created (in a folder names `css` in the same level as the `sass` folder) (again actual minification only happens if running a production build).
* Additional license and CSS map files (and rtl files) will also be created depending on whether it's a production or development build.
* If testing the builds in succession, files should be cleaned up as part of the 'clean' script (except after a production build but a development build will take care of that).
* Finally, you should be able to navigate the CRM dashboard using the built version of CRM from this branch (for example using JN and the beta plugin) with no issues.


Other things of note:
* The functions `zeroBSCRMJS_listView_generic_editdirectlink` and `zeroBSCRMJS_listView_invoice_bulkActionTitle_delete` in `ZeroBSCRM.admin.listview.js` were duplicated which broke the build (one of each removed but worth a check to see the content in the remaining one is correct as `zeroBSCRMJS_listView_generic_editdirectlink` appeared to be copy pastes from different functions).
* Also one CSS fix was needed too in `zbs-exitform.css` to add a missing opening bracket.
* There are several removed min.css (and a min.js) file where they are minified via script anyway. A few files have been unminified where only a minified version existed, to aid in the minification process via script. In the process a couple of CSS issues were resolved as well.
